### ISSUE TITLE
Building the APK

### ISSUE THREAD

**Issue by tred79** [MycroftAI/MycroftCore-Android#1]
I have downloaded the code, unzipped it, imported it into Android Studio, but every time I build either a signed or unsigned APK it fails.
![android studio](https://cloud.githubusercontent.com/assets/25230527/22120051/046d5878-de4c-11e6-8b01-fe0d6f0ed1b4.jpg)

**Comment by seymour-bootay** [comment]
I haven't done any work on this app I primary work on the "pass-through" app but typically if you get that error you need to enable multi dex.  In the build.gradle (Module: mobile) file try adding multiDexEnabled true to the default config.  So the default config section would look like this:

defaultConfig {
        applicationId "mycroft.ai.mycroftcore_android"
        minSdkVersion 21
        targetSdkVersion 24
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
    }

![image](https://cloud.githubusercontent.com/assets/1637816/22134004/9313bf58-de89-11e6-8be8-ee2aa0aae6d7.png)

**Comment by paulscott56** [comment]
Yes, thanks @seymour-bootay for that. I was just waiting for a bigger screen to reply (I hate typing on mobile).

The current NLP libs are completely inefficient for Android and are being replaced. I will try and commit the code I'm actually working on for this repo over the weekend, which is way better

**Comment by BenjaminHCCarr** [comment]
@paulscott56 I would like to try rolling the apk as well. If this and the handoff apps gain some traction it might be interesting to see a [f-droid server](https://github.com/f-droid/fdroidserver) running as well in the spirit of open source.

You mentioned that the libs in the git repo weren’t working well; but hoping to get a source bump before I dive in, as last commit was about 9 months ago, and it's been mentioned above their are known issues.

**Comment by paulscott56** [comment]
@BenjaminHCCarr reason for no commits is that I have been working on the NLP libs as a separate project. The only concern with the Stanford ones I used before were that they are very cpu intensive and chew up battery. Other than that, it works great.

An f-droid repo is a great idea! When I previously wanted to release the other app though, @spenrod said it was in review by the Mycroft team, so that was a few months back. Not sure what is happening with that now though. I'll bring it up on Slack, which I also encourage you to do!

**Comment by penrods** [comment]
Sorry, Paul and co.  I think I dropped the ball on this.

So if I understand this correctly an APK would be uploaded to the FDroid system, right?  Would we really need to run our own server?  Ideally we would have the APK generation happening via automated systems that do the upload when a new release is available.

**Comment by paulscott56** [comment]
@penrods yes, to be clear though, this would be for Mycroft-Android, not this project (yet).

https://f-droid.org/wiki/page/Known_Repositories

The F-Droid mainline repo should be good...

**Comment by penrods** [comment]
At some point our design  team wants to give you guys feedback on the look/feel of the app, but I don't want that to hold you up.  What is necessary for you to get the f-droid entry going?  You want to start the f-droid application in their Submission Queue?

**Comment by BenjaminHCCarr** [comment]
@paulscott56 is 100% right, this is perfect for mainline

my comment on own server is for projects like Guardian Project
https://github.com/guardianproject
or BRIAR:
https://blog.grobox.de/2016/briar-next-step-of-the-crypto-messenger-evolution/
Who may run their own server with different builds as options.

I would be happy to be point person with f-droid, if MyCroft is building it. If I build myself and maintain it will likely either get booted or at best labeled (unofficial). 

But I think in this case, release often, be open (thanks for saying the libs need work), and break things (IT BUILDS!!!!!), is the best approach. If we don't get more eyes on this it will wither as people won't know it exists. At the moment I am pretty sure it is dead last in the repo. If we start making Android-Core-0.0.0.1-ThisMayRunRM/ available people will be interested in testing, and subimitting issues and PR's. Like I mentioned. If I have to wornder if it is my build that is breaking it or not is a problem.

If their is a "blessed" but "less than ideal" starting point to fork/hack on/report on, then we get the ball rolling.

And while I realize that the Android-remote app to send commands is more important, their is likely overlap in bugs that will be found.

Also where is the android-remote repo?
Best,
Ben

**Comment by mvdkleijn** [comment]
Guess this initiative died out? @benjaminhccarr @penrods

**Comment by AdamDenoon** [comment]
Seems that way, @mvdkleijn

**Comment by skraynick** [comment]
@AdamDenoon are you on mattermost? We are working on a core version.

**Comment by AdamDenoon** [comment]
@skraynick I am, but admittedly I haven't kept up with it. Just logged in :)

I was answering @mvdkleijn in regards to the initiative of this particular repo.

**Comment by nicolasmaia** [comment]
So, does it build?

**Comment by Oymate** [comment]
Please just ask them and they will do it themselves
https://gitlab.com/fdroid/rfp/issues

Or you can submit the metadata yourself to the repo which will make it even faster
https://gitlab.com/fdroid/fdroiddata/blob/master/CONTRIBUTING.md

**Comment by krisgesling** [comment]
I'd be pretty wary of publishing any APK until it is ready for public use. So please don't ask F-Droid to do so. 

First it will take up their time, then it will result in a lot of support requests flowing back to us, both when we know the project isn't ready. 

We get the same thing when a distro packages Mycroft and doesn't maintain it. Works fine for a period, then we get lots of angry people saying that it's broken, and assuming that it's an official package.

I understand the desire for quick and simple Android solution but we just aren't there yet. It is one of the things we want to look at once we have shipped the Mark II though.

**Comment by Oymate** [comment]
Understood. Thanks for the communicatio. Though I do really want an android app or a STT installable at the very least :)
